cardano-cli/test/Test/Pioneers/Exercise1/BuildShelleyPaymentAddress.hs

```diff
@@ -0,0 +1,56 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Test.Pioneers.Exercise1.BuildShelleyPaymentAddress
+  ( tests
+  ) where
+
+import           Cardano.Prelude
+
+import           Hedgehog (Property, discover)
+import qualified Hedgehog as H
+
+import           Test.OptParse
+
+
+-- | This property checks the scenario where we would like to
+-- build a shelley address.
+-- 1. We generate a key pair
+-- 2. We check for the existence of the key pair
+-- 3. We use the generated verification key to build a shelley address.
+prop_buildShelleyPaymentAddress :: Property
+prop_buildShelleyPaymentAddress =
+  H.withTests 1 . H.property $ do
```

Comment:
Apart from `H.withTests` and `H.property` what are we using from HH?

Could this not just be a function in `IO Bool` or similar? Ie returns `True` on success.

Reply:
I was planning to use `Group` to group the exercises as we go along and `hedgehog` helps with the error rendering. If you are opposed to using `hedgehog` for these I can remove its usage.

Reply:
I was not so much opposed to HH as wondering why. Nice error rendering actually makes sense.

So it you want it for the error rendering you should pull out a function like:

```
propertyOnce =  H.withTests 1 . H.property
```
and then use that everywhere.